libsignal-service/src/account_manager.rs

```diff
@@ -276,6 +283,38 @@ impl<Service: PushService> AccountManager<Service> {
         Ok(())
     }
 
+    pub async fn retrieve_profile(
+        &mut self,
+        uuid: uuid::Uuid,
+    ) -> Result<Profile, ServiceError> {
+        let profile_key =
+            self.profile_key.expect("set profile key in AccountManager");
+        let profile_key = ProfileKey::create(profile_key);
+        let profile_cipher = ProfileCipher::from(profile_key);
+
+        let encrypted_profile = self
+            .service
+            .retrieve_profile_by_id(&uuid.to_string())
+            .await?;
+
+        // Profile decryption
+        let name = encrypted_profile
+            .name
+            .and_then(|data| profile_cipher.decrypt_name(data).ok()?);
```

Comment:
You're dismissing the error here. I suggest you add `ProfileCipherError` to the `ServiceError`, and then use something like

```suggestion
            .map(|data| profile_cipher.decrypt_name(data)).transpose()?;
```

(same for the other fields)

Reply:
Thanks. I misread `upload_versioned_profile`. I thought the errors are dismissed there, but they are not.